ament_package/__init__.py

```diff
@@ -16,12 +16,12 @@
 
 # set version number
 try:
-    import pkg_resources
+    import importlib_metadata
     try:
-        __version__ = pkg_resources.require('ament_package')[0].version
-    except pkg_resources.DistributionNotFound:
+        __version__ = importlib_metadata.version('ament_package')
+    except importlib_metadata.PackageNotFoundError:
         __version__ = 'unset'
     finally:
-        del pkg_resources
-except ImportError:
+        del importlib_metadata
+except (ImportError, ModuleNotFoundError):
```

Comment:
Why line can raise `ModuleNotFoundError`?

Reply:
The `import importlib_metadata` line.

Honestly, I'm not sure that catching that exception makes sense here, but I wanted to keep the same logic.

Reply:
> I wanted to keep the same logic.

Can you clarify what logic you are referring to? `ModuleNotFoundError` wasn't being catched before.

Reply:
> Can you clarify what logic you are referring to? `ModuleNotFoundError` wasn't being catched before.

The logic directly above, which is essentially:

```
try:
     import pkg_resources
except ImportError:
    __version__ = 'unset'
```

But that's Python 2 code.  In Python 3, if you try and fail to import a module, you get an `ModuleNotFoundError` instead.

Reply:
`ModuleNotFoundError` is a subclass of `ImportError` (https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError) so I don't see the point of the change.

Reply:
> `ModuleNotFoundError` is a subclass of `ImportError` (https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError) so I don't see the point of the change.

Ah, I didn't realize that.  Then you are right, we don't need that change.  I'll fix it up.